tools/generate-helpers/mockgen-wrapper

```diff
@@ -25,6 +25,10 @@ set -o pipefail
 #   This does not work when generating mocks for code that will be generated by
 #     another go:generate command.
 
+PROJECT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
+MOCKGEN_BIN="${PROJECT_ROOT}/.gotools/bin/mockgen"
+[[ -x "$MOCKGEN_BIN" ]] || make -C "$PROJECT_ROOT" "$MOCKGEN_BIN"
```

Comment:
This is a bit ugly:
- requires knowing the `./gotools/...` path
- will not DTRT if the version of mockgen changes in `go.mod`

How about introducing a `mockgen: $(MOCKGEN_BIN)` target in the `Makefile` and calling `make -C "$PROJECT_ROOT" mockgen` here **unconditionally**?

Reply:
Yeah, I'll add something like `${tool-basename}-bin` or `install-${tool-basename}` phony target.

Reply:
Why everywhere? If `which-mockgen` only depends on `$(MOCKGEN_BIN)` then you only need to make sure `>&2` is on those `@echo "+ $$(notdir $$@)"` lines in `gotools.mk`? What am I missing?
If there indeed is another reason, then I guess it's acceptable, as long as:
- it works correctly in all funny `-C` cases
- you document the reason in a comment here :-)

Reply:
Well, I guess I'd also have to apply it to the `go install` commands, no? I don't know if I can rely on the fact that they print all output to `stderr`, as there is no `stdout` output that would be usable for `$(...)` command substitution anyway.

Reply:
I suppose you won't be extremely happy in a third opinion.
I'd suggest to take it as granted that `.gotools/` is the location for binaries and binaries are expected there.
This fragment would look like this then:

```sh
MOCKGEN_BIN="${PROJECT_ROOT}/.gotools/bin/mockgen"
if [[ ! -x "$MOCKGEN_BIN" ]]; then
  die "Did not detect mockgen executable, run the following command to materialize it: make -C $PROJECT_ROOT $MOCKGEN_BIN"
fi
```

Alternatively, I'm for `which-mockgen` approach which does the same but is robust against executable renames. I think, a side effect of downloading and building something is a bit too heavy especially since that may fail e.g. due to network outage.